dev/src/blocks/pc-custom-endpoint/edit.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Retrieves the translation of text.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * React hook that is used to mark the block wrapper element.
+ * It provides all the necessary props like the class name.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/#useblockprops
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { useState, useEffect } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { decodeEntities } from '@wordpress/html-entities';
+
+import {
+	PostChooser,
+	PostChooserSidebar,
+	registerCustomEntities,
+} from '@bostonuniversity/block-imports';
+
+/**
+ * Lets webpack process CSS, SASS or SCSS files referenced in JavaScript files.
+ * Those files can contain any CSS code that gets applied to the editor.
+ *
+ * @see https://www.npmjs.com/package/@wordpress/scripts#using-css
+ */
+import './editor.scss';
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @param {Object}   props               - The component props.
+ * @param {Object}   props.attributes    - The block attributes.
+ * @param {Function} props.setAttributes - Function to update block attributes.
+ * @param {boolean}  props.isSelected    - Whether the block is currently selected.
+ * @return {Element} Element to render.
+ */
+export default function Edit( { attributes, setAttributes, isSelected } ) {
+	const {
+		selectedPostID,
+		selectedPostTitle,
+		selectedPostExcerpt,
+		selectedPostURL,
+	} = attributes;
+
+	useEffect( () => {
+		// Register the custom bob endpoint as an entity
+		const unsubscribe = registerCustomEntities( [
+			{
+				kind: 'root',
+				name: 'block-imports/v1/bob-endpoint',
+				baseURL: '/block-imports/v1/bob-endpoint',
+			},
+		] );
+
+		// Cleanup subscription on unmount
+		return () => {
+			if ( typeof unsubscribe === 'function' ) {
+				unsubscribe();
+			}
+		};
+	}, [] );
```

Comment:
I'm wondering if this should be registered globally via it's own JS file enqueued in the editor vs in the block's Edit function. If there are multiple instances of the same block on the page this gets registered and unregistered as those blocks mount I think...

If the endpoint is only used in this block maybe it's fine but something to consider perhaps.